### DISCUSSION TITLE
Задачи к следующему релизу (24.09)

### DISCUSSION THREAD

**Discussion by alkoleft** [bia-technologies/yaxunit#409] [General]
Предлагаю обсудить состав следующего релиза [24.09](https://github.com/bia-technologies/yaxunit/milestone/12)

На текущий момент состав такой:

- [ ]  #91 
- [ ] #115 
- [ ] #120 
- [ ] #290 
- [ ] #293 
- [ ] #326 
- [ ] #328 
- [ ] #394 
- [ ] #395 
- [ ] #287 
- [ ] #375 
- [ ] #381 
- [ ] #387 

Предлагайте свои варианты

**Comment by 1cgh** [comment]
Очень бы пригодилось прокидывание настроек Мокито в фоновые, в телеге обсуждали https://t.me/BIAOpenTools/12/3687, но без конкретики и вроде даже issue нет :)

**Reply by alkoleft** [reply]
Можешь это оформить в issues? Без задачи только свои хотелки делаю и то редко.

**Reply by 1cgh** [reply]
https://github.com/bia-technologies/yaxunit/issues/410

**Reply by alkoleft** [reply]
@1cgh Пока с этим есть некоторые трудности, для реализации задачи нужно еще кое-что сделать.

**Comment by alexandr-yang** [comment]
Я бы предложил еще вот эти задачи:
#382 Эмулятор объекта HTTPОтвет, будет полеpно многим при тестировании интеграции через HTTP
#349 Лично мое пожелание, хочу реализовать что-то похожее на фаззинг-тестирование на YAxUnit

**Reply by alkoleft** [reply]
`фаззинг-тестирование` новое для меня слово, я это называл стресс тестированием). Идея интересная и хочется увидеть, что из этого получится.

**Reply by 1cgh** [reply]
так и до мутационного доберемся)

**Reply by alkoleft** [reply]
Физзинг и парное, это я так понимаю больше "исследовательские" виды тестирования, не относящиеся к модульным и запускаться и обрабатываться должны отдельно. 
Нужно продумать как ими пользоваться.
Ну а пока подготовить для них базу в виде #349 и других.

Потом еще туда конструктор тестовых данных прикрутить, при этом наделив его памятью, чтобы вариативность генерировал для N-прогонов

**Reply by 1cgh** [reply]
Не совсем так. Фаззинг - это когда мы рандомную дичь шлем в метод/сервис, ожидая что он не умрет от этого, а корректно отработает (например, нормально ответит, что ты мне шлешь дичь). В 1С кстати, учитывая нестрогую типизацию, не знаю насколько это жизнеспособно.
А парное - это просто техника тест-дизайна, сокращение кейсов, которые мы хотим проверить.

**Reply by 1cgh** [reply]
Я просто не знаю, почему Александр #349 назвал фаззингом)

**Reply by alexandr-yang** [reply]
Я назвал это чем то похожим на фаззинг. И данная такска не реализует его, а лишь дает интсрумент.

Идея была такая. Если метод принимает параметры число и булево, то давайте сгенерируем разные варианты этих параметров (для числа это 0, положительное число, отрицательное, дробное и тп), передадим в метод и посмотрим упадет или нет. На сколько корректнет будет результат - не суть важно, мы не проверяем логику. 

Более жизненный пример - метод принимает перечисление и возвращает какое-то строкове значение. В коде сопоставление входящего и исходящего значения. Если значение не известно - выдает ошибку. Такое часто встречал.
И все хорошо, пока в перечислении не появится еще одно значение, а в сам метод его конечно же забывают добавлять.

**Reply by alkoleft** [reply]
Инструменты в движке нужно реализовать, а по применению - попробуем, поэкспериментируем.

**Reply by 1cgh** [reply]
> Более жизненный пример - метод принимает перечисление и возвращает какое-то строкове значение. В коде сопоставление входящего и исходящего значения. Если значение не известно - выдает ошибку. Такое часто встречал. И все хорошо, пока в перечислении не появится еще одно значение, а в сам метод его конечно же забывают добавлять.

так вот для таких случаев не очень понятно, чем поможет. что до появления нового значения перечисления нам метод возвращал ошибку, если что-то неизвестное передавали, что после, как понять разницу? я не докапываюсь, просто хочу конкретные сценарии применения увидеть
